native/src/main/java/io/ballerina/stdlib/ftp/util/FtpUtil.java

```diff
@@ -210,6 +212,16 @@ public static String getCompressedFileName(String fileName) {
         return fileName.substring(0, fileName.lastIndexOf('.')).concat(".zip");
     }
 
+    public static MethodType getOnFileChangeMethod(BObject service) {
+        MethodType[] methodTypes = service.getType().getMethods();
+        for (MethodType method: methodTypes) {
+            if (method.getName().equals(ON_FILE_CHANGE_REMOTE_FUNCTION)) {
+                return method;
+            }
+        }
+        return null;
+    }
```

Comment:
Instead of returning `null` lets use `Optional`:
```suggestion
    public static Optional<MethodType> getOnFileChangeMethod(BObject service) {
        MethodType[] methodTypes = service.getType().getMethods();
        return Stream.of(methodTypes)
                .filter(m -> ON_FILE_CHANGE_REMOTE_FUNCTION.equals(m.getName()))
                .findFirst();
    }
```

Reply:
Added in https://github.com/ballerina-platform/module-ballerina-ftp/pull/580/commits/fba816664dd0a16fedbc513a2e57237e14a6eb33